clayui.com/src/components/Editor/index.js

```diff
@@ -14,28 +15,56 @@ import theme from '../../utils/react-live-theme';
 
 const spritemap = '/images/icons/icons.svg';
 
+function formatCode(code) {
+	try {
+		return prettier.format(code, {
+			parser: 'babel',
+			plugins: [parserBabylon],
+		});
+	} catch (e) {
+		// eslint-disable-next-line
+		console.log(e);
+
+		return;
+	}
+}
+
 const Editor = ({
 	code,
 	disabled = false,
 	imports,
 	preview = true,
 	scope = {},
 }) => {
-	try {
-		code = prettier.format(code, {
-			parser: 'babel',
-			plugins: [parserBabylon],
+	let reactSnippet = {
+		code: '',
+		name: 'React',
+	};
+
+	if (Array.isArray(code)) {
+		const codeToFormat = [...code];
+
+		codeToFormat.forEach((snippet) => {
+			if (snippet.name !== 'JSP') {
```

Comment:
Instead of a strict check here, we can have include a `format` boolean in our code objects that default to true. And then for jsp or other code we don't want formatted, we just would have to pass `format: false`